Source/CombatExtended/CombatExtended/Verbs/Verb_LaunchProjectileCE.cs

```diff
@@ -307,220 +308,227 @@ public override void WarmupComplete()
                                     }
-                                    else
-                                    {
-                                        // Aim center mass
-                                        targetRange.min = victimVert.BottomHeight;
-                                        targetRange.max = victimVert.MiddleHeight;
-                                    }
-                                }
-                            }
-                            else
-                            {
-                                if ((Victim?.kindDef?.RaceProps?.Humanlike ?? false))
-                                {
-                                    if (AppropiateAimMode)
-                                    {
-                                        #region Finding highest protection apparel on legs, head and torso
-
-                                        Func<Apparel, float> funcArmor = x => (x?.GetStatValue(StatDefOf.ArmorRating_Sharp) ?? 0.1f);
+                                    #endregion
 
-                                        var Torso = Victim.health.hediffSet.GetNotMissingParts().Where(X => X.IsCorePart).First();
 
-                                        var TorsoArmors = (Victim?.apparel?.WornApparel?.FindAll(F => F.def.apparel.CoversBodyPart(Torso)) ?? null);
+                                }
+                                else // Shooting at non-humanlike, go for headshots
+                                {
+                                    targetRange.min = victimVert.MiddleHeight;
+                                    targetRange.max = victimVert.Max;
+                                }
+                            }
 
-                                        Apparel TorsoArmor = TorsoArmors.MaxByWithFallback(funcArmor);
+                        }
 
-                                        var Head = Victim.health.hediffSet.GetNotMissingParts().Where(X => X.def.defName == "Head").FirstOrFallback(Torso);
+                    }
 
-                                        var Helmets = (Victim?.apparel?.WornApparel?.FindAll(F => F.def.apparel.CoversBodyPart(Head)) ?? null);
 
-                                        Apparel Helmet = Helmets.MaxByWithFallback(funcArmor);
+                }
 
-                                        var Leg = Victim.health.hediffSet.GetNotMissingParts().Where(X => X.def.defName == "Leg").FirstOrFallback(Torso);
+                targetHeight = targetRange.Average;
+                if (projectilePropsCE != null)
+                {
+                    targetHeight += projectilePropsCE.aimHeightOffset;
+                }
+                if (targetHeight > CollisionVertical.WallCollisionHeight && roofed)
+                {
+                    targetHeight = CollisionVertical.WallCollisionHeight;
+                }
+            }
+            return targetHeight;
+        }
+        protected virtual bool LockRotationAndAngle => numShotsFired > 0;
```

Comment:
nit: the previous name `MidBurst` is I think more descriptive since this property is used also in shootline checks where it has nothing to do with ballistics. Also this should move up to live alongside the other property declarations.

Reply:
Now there is MidBurst property that used in LockRotationAndAngle by default